Forms: Add slider field

edanzer opened the pull request:
Addresses [FORMS-66](https://linear.app/a8c/issue/FORMS-66/forms-rangeslider-field)

## Proposed changes:

Adds a new slider field block for forms. The goal of this PR is not a final, releasable block. Rather we want to get this to 'good enough' state that we can merge it as a beta block, and then do iterative improvements as needed until it is ready for release. 

This is based on designs from this P2: p1HpG7-wSU-p2

Changes include: 
- Slider field block (the parent block for the field, which contains a label and a slider input block)
- Slider input block (the actual slider, which use a browser native range input)
- Like other blocks it uses the Jetpack Field component, which has been updated to accommodate slider inputs as a child. 
- Other standard boiler plate to handle block registration and frontend render callbacks
- Users interactivity API for front end behavior

This is based on designs in this P2: p1HpG7-wSU-p2

**Screenshot: slider field in editor**
<img width="1239" height="583" alt="slider-field-editor" src="https://github.com/user-attachments/assets/4264ddde-f703-4bc9-bbc0-41095e64de9a" />

**Screenshot: slider field on frontend**
<img width="719" height="129" alt="slider-field-frontend" src="https://github.com/user-attachments/assets/f5081f51-a778-41b2-9ff1-98dc699c4c60" />

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
[FORMS-66](https://linear.app/a8c/issue/FORMS-66/forms-rangeslider-field)

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

**Confirm block is hidden by default.** Confirm that if you do not have beta blocks enabled, the block is not visible. This is most important, as other improvements to the block itself can be made after the block is merged as a beta block. 

**Test block**.
* Enable beta blocks: `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`
* Note: You may need to run `jetpack build packages/forms` to get the frontend slider-field.css to be built and loaded in the dist directory. Having the watch command running is not sufficient.
* Add a form to a page or post
* Add a new Slider block. It should contain a label and range input as children. 
* Confirm it looks like the editor screenshot above by default. 
* Publish and test the frontend based on the default. Confirm it looks like screenshot above on frontend.
* Submitting your form and confirm that it submits data correctly. 
* Try playing with the min, max, and starting values. Save and test frontend. Confirm they apply as expected.

edanzer committed:
Add Forms slider field block

edanzer committed:
changelog

edanzer committed:
Misc changes in progress

edanzer committed:
Many misc changes still in progress

edanzer committed:
Improve frontend slider styling

edanzer committed:
Fix and improve editor rendering

edanzer committed:
Correctly set frontend default values

enejb commented on projects/packages/forms/src/modules/slider-field/view.js:
```diff
@@ -0,0 +1,33 @@
+
+import { store, getContext } from '@wordpress/interactivity';
+
+store( 'jetpack/field-slider', {
```
Instead of creating a new `'jetpack/field-slider'` slider specific store lets reuse the one that is already available for the field via `jetpack/form`. This will make sure that the form has at least one field set in the form. As well as take care of the field validation. (Know when to display the error messages)

edanzer replied:
This has been updated!

edanzer committed:
Update slider to use shared form context

edanzer committed:
Remove unneeded context

edanzer committed:
Fix slide indicator positioning

edanzer committed:
Add proper slider icon for field and input

edanzer committed:
Rename slider-input to input-slider

edanzer committed:
Rename slider input to range input

edanzer committed:
Update Slider field to sentence case

edanzer committed:
Fix range input name in frontend render callback

edanzer committed:
Fix handling of starting value

enejb commented on projects/packages/forms/src/contact-form/class-contact-form-field.php:
```diff
@@ -2169,4 +2176,109 @@ function ( $key, $value ) {
+		</div>
+		<?php
+		$field .= ob_get_clean();
+		return $field;
```
I think you will want to also do add `$this->get_error_div( $id, 'slider' ); ` So that error message show up. As expected. Although I feel like it is really hard to get into an error state with a slider.

```
return $field . $this->get_error_div( $id, 'slider' );
```

enejb commented on projects/packages/forms/src/contact-form/class-contact-form-field.php:
```diff
@@ -2169,4 +2176,109 @@ function ( $key, $value ) {
+		?>
+		<div
+			class="jetpack-slider-input-row"
+			data-wp-interactive="jetpack/form"
```
I am not sure this line is necessery.

edanzer replied:
@enejb - for interactivity API, I guess since we have `data-wp-interactive="jetpack/form"` on the main form container, and we're using the same store, it'll just pick up from that?

edanzer replied:
Just confirmed it's fine to remove this and pushed.

enejb commented on projects/packages/forms/src/contact-form/class-contact-form-field.php:
```diff
@@ -2169,4 +2176,109 @@ function ( $key, $value ) {
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- output is pre-escaped by method
+			echo wp_interactivity_data_wp_context(
+				array(
+					'sliderValue' => $current_value,
```
Can we use the `fieldValue` as the slider value instead. This is already being padded into the context. In line 
https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/contact-form/class-contact-form-field.php#L1870

$value I think you can set the $value in two ways. One via the URL query parameter. And the second way is via the `default` attribute on the block. I haven't tried this yet but that worked for the ratings block. See https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/blocks/field-rating/index.js#L27

edanzer replied:
@enejb - I just pushed some change and I think I have this wired the way you're looking for now. 

In the editor I no longer distinguish between the active slider value and the 'starting' value. If you set the slider at a particular number, that will also be the starting value. I'm using 'default' in the editor to do that. 

On the frontend, I've updated to use fieldValue and removed some redundant interactivity directives. 

Feel free to have a look and let me know if you see outstanding things need for all the basic value / default value / interactivity wiring.

enejb commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,96 @@
+.jetpack-slider-input-row input[type="range"] {
+	-webkit-appearance: none;
+	appearance: none;
+	height: 2px;
```
I wish this was larger so that things would be easier to slide around. Click it easier.

edanzer replied:
I bumped the sizes for the line and bullet for the range slider.

simison commented on projects/packages/forms/src/blocks/input-range/editor.scss:
```diff
@@ -0,0 +1,95 @@
+}
+
+.jetpack-input-range__range {
+	-webkit-appearance: none;
```
Sometimes these vendor-prefixed keys get added by the CSS build pipeline based on `appearance`, so worth checking if we need to add this. Minor thing to leave it, too.

simison commented on projects/packages/forms/src/blocks/input-range/edit.js:
```diff
@@ -0,0 +1,93 @@
+				</PanelBody>
+			</InspectorControls>
+			<div { ...blockProps }>
+				<div className="jetpack-input-range-row">
```
For file field we were following this pattern: `jetpack-form-file-field__dropzone` so might make sense here to stay similar. Not sure what other fields use.

edanzer replied:
Yup, I updated this inline with others to use jetpack-field-slider__ as the prefix.

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,96 @@
+	align-items: center;
+	width: 100%;
+	gap: 12px;
+	margin-top: 30px;
```
These can probably use some "default block gap" variable so that the gap between different fields stays consistent across themes.

edanzer replied:
This actually isn't spacing for the block itself. We are adding an 'indicator' that shows the current selected value. It loves above the range input, and we need to have spacing to accommodate that. 

<img width="832" height="151" alt="range-inpute-top-margin" src="https://github.com/user-attachments/assets/330eae37-6816-4ac8-abf9-ff375607f958" />

simison commented on projects/packages/forms/src/blocks/input-range/editor.scss:
```diff
@@ -0,0 +1,95 @@
+
+.jetpack-input-range__min-label,
+.jetpack-input-range__max-label {
+	font-size: .9em;
```
Would be good to opt-into typography settings for slider and re-use default theme font size, yet allow customization.

edanzer replied:
There are multiple instances of text in the slider designs:

<img width="616" height="215" alt="slider-text" src="https://github.com/user-attachments/assets/f6b6453e-41b2-42d9-a357-791413fa0b5d" />

A question is if we opt the slider field into typography settings, should those settings apply to all? It probably makes sense to apply them to the numeric values and labels that appear on the left/right of the input. The small value indicator on top is trickier, because it's inversed with background and text.

simison commented on projects/packages/forms/src/blocks/input-range/editor.scss:
```diff
@@ -0,0 +1,95 @@
+	height: 2px;
+	background: var(--jetpack--contact-form--button-primary--background-color);
+	outline: none;
+	border-radius: 2px;
```
Could be we opt-in to support theme inherited `border-radius`. Not sure we should default to border radius from buttons since the elements are so different size, but that would allow more boxy looking sliders on boxy looking themes. 🤔 Anyway, we could allow customizing this in the sidebar.

edanzer replied:
This border radius is actually for the very small line of the range input. I'm not sure default border radius for other elements makes sense there? And it seems like a very small element to allow customization?

<img width="714" height="140" alt="slider-input-border-radius" src="https://github.com/user-attachments/assets/16023f88-caf1-4b19-b5c0-e4e6524b71b2" />

simison replied:
Gotcha! 

Yeah it probably makes more sense with the badge, but more customization options are fine for a follow-up.

simison replied:
And overall yes, it'll make sense to allow more boxy designs and rounded designs but we'll need to figure out how to do it:

<img width="602" height="195" alt="image" src="https://github.com/user-attachments/assets/597d5195-1bf3-4d4a-9d1b-8933f4c5333b" />

([via](https://www.w3schools.com/howto/howto_js_rangeslider.asp))

simison commented on projects/packages/forms/src/blocks/input-range/edit.js:
```diff
@@ -0,0 +1,93 @@
+			</InspectorControls>
+			<div { ...blockProps }>
+				<div className="jetpack-input-range-row">
+					<span className="jetpack-input-range__min-label">{ min }</span>
```
Adding `contenteditable="plaintext-only"` feels pretty good here for allowing editing values directly in the canvas. It'll need some code to allow only digits as values and defaulting to 0 for empty content, though.

Alternatively `<input type="number">` but that needs overriding all styles for the input to blend it in, but gives you up/down arrow controls for free, number keyboard on mobile, etc. nice things.

I'd probably go number input route myself.

edanzer committed:
Include error div, remove interactivity container

edanzer committed:
Update css class namespace

edanzer committed:
Consolidate and update frontend slider styles

edanzer committed:
Update classnames on backend

edanzer committed:
Consolidate and tweak editor styles

edanzer commented:
> Using `accent-color` in CSS might help retain the "unfilled" background right from the drag handle
> 

I played with this for a long time. There are some complexities. It may still be doable, but I'd like to get this merged as is first as a beta block, and then work on this change separately.

edanzer committed:
Make slider block a beta block

edanzer committed:
Update to use default value in editor/frontend

edanzer committed:
Tweak interactivity api logic

edanzer commented on projects/packages/forms/src/blocks/contact-form/child-blocks.js:
```diff
@@ -51,7 +53,9 @@ export const childBlocks = [
-	...( getJetpackBlocksVariation() === 'beta' ? [ JetpackRatingField, JetpackRatingInput ] : [] ),
+	...( getJetpackBlocksVariation() === 'beta'
+		? [ JetpackRatingField, JetpackRatingInput, JetpackFieldSlider, JetpackSliderInput ]
+		: [] ),
```
The block is now a beta block.

edanzer committed:
Update block examples to use default

edanzer committed:
Small cleanup

CGastrell reviewed (dismissed):
Looks good! And works as described.

The only case I was able to confuse the field was setting max and min settings the same value and somehow I was able to set the default to 0. Ends up in an inoperable field on the frontend, but that's up to the user to not mess up.

In any case, being beta we can follow up on any weird case we want to address

simison commented on projects/packages/forms/src/blocks/input-range/index.js:
```diff
@@ -0,0 +1,45 @@
+		},
+		default: {
+			type: 'number',
+			default: 50,
```
Wondering it would make sense to not be opinionated about default being right in the middle, and default this also to 0?

edanzer replied:
Updated this to 0.

simison commented on projects/packages/forms/src/blocks/shared/components/jetpack-field.js:
```diff
@@ -34,7 +34,7 @@ const JetpackField = props => {
 		return [
 			[ 'jetpack/label', { label, required, requiredText } ],
-			[ 'jetpack/input', { type } ],
+			[ type === 'slider' ? 'jetpack/input-range' : 'jetpack/input', { type } ],
```
Instead of abstracting at this file, other more custom blocks abstract directly at `edit.js`:

https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/blocks/field-file/edit.js
https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/blocks/field-rating/edit.js

It results a bit more boiletplate (which we can abstract to hooks of course) but otherwise wondering if we should stick to one consistent way?

edanzer replied:
It seems like there are a few approaches we could take. 

1. Most of our fields use JetpackField, which then outputs the default Label and Input blocks. If custom logic is needed, it is handled in the Input block. For example, the Input block conditionally outputs a rich text component for dropdowns or conditionally loads [min/max inspector controls](https://github.com/Automattic/jetpack/blob/1e9cef5af36c1d4b7169c2e54d5dfb50b4aa5667/projects/packages/forms/src/blocks/input/edit.js#L82) for number inputs. 
   => I could do this with the slider input. I could remove the code you flagged here and update Input to  output slider-specific markup and inspector controls if type = slider. 
   => The upside is we re-use the same components as much as possible, we tweak the Input block in ways that we're already doing, we don't have to tweak Jetpack Field, and I'd be able to drop the custom input-range component I added entirely. 
   => The downside is we stretch the Input block by adding a lot of field-specific logic. 
  
2. On other side, the two you mention do not use JetpackField or the default Input block at all. They directly load their own children. This makes sense because they both load very custom inputs where using the Input block doesn't make much sense.  
   => The upside is we don't don't over extend JetpackField or Input.
   => The downside is that JetpackField and Input offer a lot of built in logic for free that you need to take care to duplicate. 

3. I was trying a middle approach that uses JetpackField, but tweaks it to output a separate input. 
    => The upside is I still get all the JetpackField logic for free, and I don't overly stretch the Input block. 
    => The downside is we have field-specific logic in JetpackField now, I don't re-use the Input block, I have to create a custom input-range component, and finally this hybrid approach differs from other blocks which have generally taken option 1 or 2. 

Thoughts?

simison replied:
Nice summay of cons/pros! Thanks.

> => The downside is that JetpackField and Input offer a lot of built in logic for free that you need to take care to duplicate.

I'm less worried about boilerplate/duplicate logic if it results in cleaner if/else structure of code. I.e. code shouldn't feel too mangled up, and unclear when you re-use a component. What actually happens is that the component contains if/else logic at the deepest level only and does not contain an if/else fork somewhere halfway through the component.

It's hard to back out from mangled code, but easy to remove boilerplate by moving things to hooks and functions.

How would the decision look if we contained a bit more the boilerplate to avoid it? Might be helpful regardless for file uploads and ratings blocks, too.

> => The upside is I still get all the JetpackField logic for free, and I don't overly stretch the Input block.

> or conditionally loads [min/max inspector controls](https://github.com/Automattic/jetpack/blob/1e9cef5af36c1d4b7169c2e54d5dfb50b4aa5667/projects/packages/forms/src/blocks/input/edit.js#L82) for number inputs.

Yeah, I'm not sure if this is great either code-wise and we could probably move them a bit up. I get they ended up there probably to have the settings in the number input rather than field.

edanzer replied:
I've refactored this field to avoid JetpackField and use custom edit components instead. It largely follows the patterns of the Rating Field and Input now.

simison commented on projects/packages/forms/src/blocks/shared/util/constants.js:
```diff
@@ -1,7 +1,7 @@
 
 export const ALLOWED_FORMATS = [ 'core/bold', 'core/italic' ];
-export const ALLOWED_INNER_BLOCKS = [ 'jetpack/label', 'jetpack/input' ];
+export const ALLOWED_INNER_BLOCKS = [ 'jetpack/label', 'jetpack/input', 'jetpack/input-range' ];
```
Should this be allowed here or only as inner block of `field-range`?

edanzer replied:
This was added when we were using JetpackField for the slider field. I've refactored away from that, and so have remove this change too.

simison commented:
Overall looking good and file to merge for follow-up iterations.

Multiple sliders in single form or multiple forms with sliders in single page are causing trouble:

<img width="789" height="399" alt="Screenshot 2025-07-21 at 11 19 57" src="https://github.com/user-attachments/assets/9f6ad54a-20c5-4c7d-97d3-4f0250594dfb" />
<img width="822" height="494" alt="Screenshot 2025-07-21 at 11 20 27" src="https://github.com/user-attachments/assets/0e8694d3-c56a-4043-bbdf-c4445c8ee5b6" />

When changing max value be smaller than starting value, it feels fine to allow it "on typing", as they might be changing it by removing e.g. zeroes from "100" so it temporarily becomes "1". But when unfocusing and leaving the number mismatched, it might make sense to either:

- Set default to the same as max
- Highlight default as errored so that they notice the mistake

<img width="1477" height="611" alt="Screenshot 2025-07-21 at 11 21 21" src="https://github.com/user-attachments/assets/96fb4c7a-f492-4c08-92f7-fd9508e27204" />

enejb commented on projects/packages/forms/src/modules/slider-field/view.js:
```diff
@@ -0,0 +1,44 @@
+ */
+function getSliderMinMax( context ) {
+	const root = context._rootNode || document;
+	const input = root.querySelector( 'input[type="range"]' );
```
instead of querying the dom could we not pass in the min and max value via the context? 
via `data-wp-context='{ "foo": "bar" }' ` see https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/api-reference/#wp-context

edanzer replied:
Yep, I've updated this!

edanzer committed:
Refactor to use custom editor components

edanzer committed:
Remove unneeded changes after refactor

edanzer committed:
Move useEffect down

edanzer committed:
Change slider default to 0

edanzer committed:
Update frontend min/max to use context

edanzer commented:
Rebased and made the following changes: 

**EDITOR**
- Refactored from using JetpackField to using custom edit.js components plus JetpackControls. This follows a similar pattern to the rating field now. Because we're not using JetpackField, I removed a few changes that were related to that. 
- I moved the min/max/default attributes and methods from input-slider to the parent field-slider block. These values, and the callback to change default, are now passed to the input via context. 
- Updated default value from 50 to 0. 
- Moving the slider now changes the default value
- Updated CSS prefixes

**FRONTEND**
- Use jetpack/form for interactivity store (vs jetpack/field-slider)
- Fetch min/max values via interactivity context (vs JavaScript)
- Use fieldValue for value, which is set in context for all fields via shared code
- Handle min/max/default attributes coming from the parent field vs input
- Update default to 0
- Output error div correctly
- Updated CSS prefixes

**DEFER TO FOLLOW UPS**
- Add options to control font size/color for various elements
- Separately style accent-color (this is tricky and not strictly needed)
- Update min/max values directly from editor (also tricky and not strictly needed)
- Handle multiple slider on a page

edanzer committed:
Fix linter errors

simison commented on projects/packages/forms/src/blocks/input-range/editor.scss:
```diff
@@ -0,0 +1,81 @@
+	padding: 2px 8px;
+	font-size: 0.7em;
+	z-index: 2;
+} 
```
Fix linter problem:

```suggestion
}

```

edanzer replied:
Fixed.

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,82 @@
+}
+
+.jetpack-field-slider__input-container input[type="range"] {
+	-webkit-appearance: none;
```
Nit: I bet our build-pipeline adds this `-webkit-appearance` here automatically because there already is `appearance`.

You could check the built CSS to confirm.

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,82 @@
+	border-radius: 50%;
+	cursor: pointer;
+	transition: background 0.3s;
+	box-shadow: 0 0 2px rgba(0,0,0,0.2);
```
Make linter happy:

```suggestion
	box-shadow: 0 0 2px rgba(0, 0, 0, 0.2);
```

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,82 @@
+	box-sizing: border-box;
+}
+
+.jetpack-field-slider__input-container input[type="range"]:focus {
```
These and the few following styles could be inside the parent — kind of personal preference but feels it makes it easier to read:

```scss
.jetpack-field-slider__input-container input[type="range"] {
	&:focus { … }
	&::-webkit-slider-thumb { … }
	&::-moz-slider-thumb { … }
	…
}
```

simison replied:
I realised the file is CSS, not SCSS. Fine as is then but since we build CSS for minification anyway, using SCSS would be fine in follow-ups. :-)

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,82 @@
+.jetpack-field-slider__input-container input[type="range"]::-moz-range-thumb,
+.jetpack-field-slider__input-container input[type="range"]::-ms-thumb {
+	width: 16px;
+	height: 16px;
```
Would be good to see shared styles grouped together with `-webkit-slider-thumb` so that they're not duplicated, making it harder to miss making changes to both in future.

Any differend style handling can be handled in separate style blocks.

simison commented on projects/packages/forms/src/contact-form/css/slider-field.css:
```diff
@@ -0,0 +1,82 @@
+	transition: background 0.3s;
+}
+
+.jetpack-field-slider__input-container input[type="range"]:focus::-webkit-slider-thumb,
+.jetpack-field-slider__input-container input[type="range"]:focus::-moz-range-thumb,
+.jetpack-field-slider__input-container input[type="range"]::-ms-fill-lower,
+.jetpack-field-slider__input-container input[type="range"]::-ms-fill-upper,
+.jetpack-field-slider__input-container input[type="range"]:focus::-ms-fill-lower,
+.jetpack-field-slider__input-container input[type="range"]:focus::-ms-fill-upper {
```
Feels like this could be more readable when nested:
```scss
.jetpack-field-slider__input-container input[type="range"] {
    &:focus::-webkit-slider-thumb,
    &:focus::-moz-range-thumb,
    &::-ms-fill-lower,
    &::-ms-fill-upper,
    &:focus::-ms-fill-lower,
    &:focus::-ms-fill-upper {
        …
    }
}
```

Or perhaps even so, but I wasn't sure if the order of `-ms*` rules was important in which case this wouldn't work:

```scss
.jetpack-field-slider__input-container input[type="range"] {
    &:focus {
        &::-webkit-slider-thumb,
        &::-moz-range-thumb,
        &::-ms-fill-lower,
        &::-ms-fill-upper {
            …
        }
    }
    &::-ms-fill-lower,
    &::-ms-fill-upper {
        …
    }
}
```

simison replied:
I realised the file is CSS, not SCSS. Fine as is then but since we build CSS for minification anyway, using SCSS would be fine in follow-ups. :-)

simison commented on projects/packages/forms/src/blocks/field-slider/edit.js:
```diff
@@ -0,0 +1,122 @@
+						value={ min }
+						onChange={ updateMin }
+					/>
+					<NumberControl
```
Let's opt-in to edge styles like in rest of the blocks: `__next40pxDefaultSize`

simison replied:
Pushed in https://github.com/Automattic/jetpack/pull/44150/commits/2aae3d7cc660f1ed3e371c4f0744eaebf2c646e8

simison commented:
When inserting field outside form (1.), it doesn't wrap itself into a form block and lacks styling (2.). It continues to work when added inside form block (3.):

<img width="1034" height="613" alt="Screenshot 2025-08-01 at 14 13 38" src="https://github.com/user-attachments/assets/0b16d3c8-97d4-404a-b85a-19989bf670cd" />

CGastrell commented on projects/packages/forms/src/blocks/field-slider/edit.js:
```diff
@@ -0,0 +1,133 @@
+import { __ } from '@wordpress/i18n';
+import JetpackFieldControls from '../shared/components/jetpack-field-controls';
+
+export default function SliderFieldEdit( props ) {
```
```suggestion
export default function SliderFieldEdit( props ) {
	useFormWrapper( props );
```
Adding that line will force a form whenever the slider is directly inserted on a post

simison reviewed (dismissed):
Tests well, I think this is good to merge behind beta for more testing and most importantly fixing the multiple sliders -case.
